connectors/services/sync.py

```diff
@@ -162,26 +115,80 @@ async def _run(self):
 
         logger.info(f"Service started, listening to events from {es_host}")
         try:
-            while self.running:
-                try:
-                    logger.debug(f"Polling every {self.idling} seconds")
-                    query = self.connectors.build_docs_query(
-                        native_service_types, connectors_ids
-                    )
-
-                    async for connector in self.connectors.get_all_docs(query=query):
-                        await self._one_sync(connector, es, sync_now)
-                    if one_sync:
-                        break
-                except Exception as e:
-                    logger.critical(e, exc_info=True)
-                    self.raise_if_spurious(e)
-                finally:
-                    if one_sync:
-                        break
-                await self._sleeps.sleep(self.idling)
+            if one_sync:
+                logger.debug(f"Running a single sync")
+                await self._tick()
+            else:
+                while self.running:
+                    try:
+                        logger.debug(f"Polling every {self.idling} seconds")
+                        await self._tick()
+                        await self._sleeps.sleep(self.idling)
+                    except Exception as e:
+                        logger.critical(e, exc_info=True)
+                        self.raise_if_spurious(e)
         finally:
             self.stop()
             await self.connectors.close()
-            await es.close()
         return 0
+
+    async def _tick(self):
+        native_service_types = self.config.get("native_service_types", [])
+        if "connector_id" in self.config:
+            connectors_ids = [self.config.get("connector_id")]
+        else:
+            connectors_ids = []
+
+        logger.debug(f"Native support for {', '.join(native_service_types)}")
+
+        query = self.connectors.build_docs_query(native_service_types, connectors_ids)
+
+        async for connector in self.connectors.get_all_docs(query=query):
+            await connector.heartbeat()
```

Comment:
It seems the `heartbeat` simply updates `last_seen` without checking when it is last seen.

Reply:
I think it's okay?

Protocol says that "last_seen" should be updated at least every 30 minutes, but it's "at least" - it does not mean that it has to be updated exactly every 30 minutes, so technically we can update it every second, minute, 10 minutes, etc.